src/node.rs

```diff
@@ -462,195 +491,225 @@ impl InMemoryNode {
             inner.current_batch += 1;
             inner.current_miniblock += 1;
         }
+
+        Ok(())
     }
 }
 
 impl EthNamespaceT for InMemoryNode {
+    /// Returns the chain ID of the node.
     fn chain_id(&self) -> jsonrpc_core::BoxFuture<jsonrpc_core::Result<zksync_basic_types::U64>> {
-        let inner = self.inner.read().unwrap();
-        Ok(U64::from(inner.fork_storage.chain_id.0 as u64)).into_boxed_future()
+        match self.inner.read() {
+            Ok(inner) => Ok(U64::from(inner.fork_storage.chain_id.0 as u64)).into_boxed_future(),
+            Err(_) => Err(into_jsrpc_error(Web3Error::InternalError)).into_boxed_future(),
+        }
     }
-
+    /// Calls the specified function on the L2 contract with the given arguments.
```

Comment:
Can we please keep the newlines between functions? I find it much easier to read that way